File: backend/src/main/java/com/SeeTohJJ/Backend/common/exception/InsufficientEnergyException.java
package com.SeeTohJJ.Backend.common.exception;

public class InsufficientEnergyException extends RuntimeException {

    private final int currentEnergy;
    private final int requiredEnergy;
    private final long secondsUntilNextEnergy;

    public InsufficientEnergyException(
            int currentEnergy,
            int requiredEnergy,
            long secondsUntilNextEnergy) {

        super("Not enough energy");

        this.currentEnergy = currentEnergy;
        this.requiredEnergy = requiredEnergy;
        this.secondsUntilNextEnergy = secondsUntilNextEnergy;
    }

    public int getCurrentEnergy() {
        return currentEnergy;
    }

    public int getRequiredEnergy() {
        return requiredEnergy;
    }

    public long getSecondsUntilNextEnergy() {
        return secondsUntilNextEnergy;
    }
}

File: frontend/lib/features/study/exceptions/insufficient_energy_exception.dart
class InsufficientEnergyException implements Exception {
  final int currentEnergy;
  final int requiredEnergy;
  final int secondsUntilNextEnergy;

  InsufficientEnergyException({
    required this.currentEnergy,
    required this.requiredEnergy,
    required this.secondsUntilNextEnergy,
  });
}